### PR TITLE
add connect_timeout and read_timeout configuration

### PR DESCRIPTION
**Pull request by pushrax**
`requests` by default has no timeout, which isn't a sane configuration for our usecase.

The established pattern for configuration at the moment is to use a global in the `turbopuffer` module, which isn't optimal but works.

### CODE REVIEW

File: turbopuffer/__init__.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 api_base_url = os.environ.get('TURBOPUFFER_API_BASE_URL', 'https://api.turbopuffer.com')
 upsert_batch_size = 10_000
 max_retries = 6
+connect_timeout = 10 # seconds
+read_timeout = 180 # seconds
```
Comment (morgangallant) [inline turbopuffer/__init__.py:8]: seems a bit high?

**Review by morgangallant — APPROVED** [review]

File: turbopuffer/__init__.py (hunk 1 above)
Comment (pushrax) [reply turbopuffer/__init__.py:8]: agreed, but I'd rather have some wiggle room for users who send gbps of data with a lot of concurrent requests, which can take a long time to process

**Comment by sirupsen** [comment]
@pushrax will this retry? 👍🏻

**Comment by pushrax** [comment]
> will this retry?

as of https://github.com/turbopuffer/turbopuffer-python/pull/37 yes

**Pull request merged**